src/types/UiaTextRangeBase.cpp

```diff
@@ -310,14 +310,319 @@ IFACEMETHODIMP UiaTextRangeBase::ExpandToEnclosingUnit(_In_ TextUnit unit) noexc
+
+        // The underline color is stored as a COLORREF.
+        // However, the text buffer doesn't actually store the color.
+        // Instead, we just store whether or not the text is underlined.
+        // So we'll assume that _any_ query here (except 0) is interpreted as "is the
+        // text underlined?"
+
+        if (val.iVal == 0)
+        {
+            return [](const TextAttribute& attr) {
+                return !attr.IsUnderlined();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) {
+                return attr.IsUnderlined();
+            };
+        }
+    }
+    case UIA_UnderlineStyleAttributeId:
+    {
+        // Expected type: VT_I4
+        THROW_HR_IF(E_INVALIDARG, val.vt != VT_I4);
+
+        // The underline style is stored as a TextDecorationLineStyle.
+        // However, The text buffer doesn't have different styles for being underlined.
+        // Instead, we just store whether or not the text is underlined.
+        // So we'll assume that _any_ query here is interpreted as "is the
+        // text underlined?" UNLESS we are asked for TextDecorationLineStyle_None
+        const TextDecorationLineStyle queryUnderlineStyle{ static_cast<TextDecorationLineStyle>(val.iVal) };
+        if (queryUnderlineStyle == TextDecorationLineStyle_None)
+        {
+            return [](const TextAttribute& attr) {
+                return !attr.IsUnderlined();
+            };
+        }
+        else
+        {
+            return [](const TextAttribute& attr) {
+                return attr.IsUnderlined();
+            };
+        }
+    }
+    default:
+        return nullptr;
+    }
+}
+
+IFACEMETHODIMP UiaTextRangeBase::FindAttribute(_In_ TEXTATTRIBUTEID attributeId,
+                                               _In_ VARIANT val,
+                                               _In_ BOOL searchBackwards,
+                                               _Outptr_result_maybenull_ ITextRangeProvider** ppRetVal) noexcept
+{
+    RETURN_HR_IF(E_INVALIDARG, ppRetVal == nullptr);
+    *ppRetVal = nullptr;
+
+    // AttributeIDs that require special handling
+    switch (attributeId)
+    {
+    case UIA_FontNameAttributeId:
+    {
+        RETURN_HR_IF(E_INVALIDARG, val.vt != VT_BSTR);
+        const std::wstring queryFontName{ val.bstrVal };
+        if (queryFontName == _pData->GetFontInfo().GetFaceName())
+        {
+            Clone(ppRetVal);
+        }
+        UiaTracing::TextRange::FindAttribute(*this, attributeId, val, searchBackwards, static_cast<UiaTextRangeBase&>(**ppRetVal));
+        return S_OK;
+    }
+    case UIA_FontSizeAttributeId:
+    {
+        RETURN_HR_IF(E_INVALIDARG, val.vt != VT_R8);
+        const auto queryFontSize{ val.dblVal };
+
+        // TODO CARLOS: how do I get the font size in points?
+        if (queryFontSize == 0) //_pData->GetFontInfo().GetUnscaledSize())
+        {
+            Clone(ppRetVal);
+        }

```

Comment:
@DHowett or @miniksa Any ideas on how to get the font size in points from `FontInfo`?

Reply:
You mean instead of pixels? I believe in the dx renderer is a conversion. You could move it to a helper

Reply:
I'm gonna punt on this one and file it as a follow-up because...
1. we don't expect the font size to change within the buffer, so exposing it isn't urgent
2. I'll have to do a different approach between ConHost and Windows Terminal here. And that just sounds like extra work for an already bulky PR.